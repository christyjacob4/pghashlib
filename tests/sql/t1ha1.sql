-- Test basic hash functionality with text
SELECT t1ha1('hello world');

-- Test hash with different text inputs
SELECT t1ha1('test string');
SELECT t1ha1('another test');

-- Test text input with custom seed
SELECT t1ha1('hello world', 42);
SELECT t1ha1('hello world', 84);

-- Test bytea input
SELECT t1ha1('hello world'::bytea);

-- Test bytea input with custom seed
SELECT t1ha1('hello world'::bytea, 42);

-- Test integer input
SELECT t1ha1(12345);
SELECT t1ha1(-12345);

-- Test integer input with custom seed
SELECT t1ha1(12345, 42);
SELECT t1ha1(-12345, 84);

-- Test consistency (same input should give same hash)
SELECT t1ha1('consistent test') = t1ha1('consistent test');

-- Test seed effect (same input, different seeds should give different hashes)
SELECT t1ha1('seed test', 1) != t1ha1('seed test', 2);

-- Test empty string
SELECT t1ha1('');

-- Test single character
SELECT t1ha1('a');

-- Test long string
SELECT t1ha1('Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.');

-- Test various input lengths to test different code paths
SELECT t1ha1('1234567');           -- 7 bytes
SELECT t1ha1('12345678');          -- 8 bytes  
SELECT t1ha1('123456789012345');   -- 15 bytes
SELECT t1ha1('1234567890123456');  -- 16 bytes
SELECT t1ha1('12345678901234567890123456789012'); -- 32 bytes
SELECT t1ha1('123456789012345678901234567890123'); -- 33 bytes

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 't1ha1'
ORDER BY proname, proargtypes;

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';