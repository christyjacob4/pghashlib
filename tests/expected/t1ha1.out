-- Test basic hash functionality with text
SELECT t1ha1('hello world');
        t1ha1         
----------------------
 -3912648815814528862
(1 row)

-- Test hash with different text inputs
SELECT t1ha1('test string');
       t1ha1       
-------------------
 22836396329573165
(1 row)

SELECT t1ha1('another test');
        t1ha1        
---------------------
 8532611023766273279
(1 row)

-- Test text input with custom seed
SELECT t1ha1('hello world', 42);
        t1ha1        
---------------------
 6512687883200614590
(1 row)

SELECT t1ha1('hello world', 84);
        t1ha1        
---------------------
 8672737372929091947
(1 row)

-- Test bytea input
SELECT t1ha1('hello world'::bytea);
        t1ha1         
----------------------
 -3912648815814528862
(1 row)

-- Test bytea input with custom seed
SELECT t1ha1('hello world'::bytea, 42);
        t1ha1        
---------------------
 6512687883200614590
(1 row)

-- Test integer input
SELECT t1ha1(12345);
        t1ha1         
----------------------
 -3263203806586990028
(1 row)

SELECT t1ha1(-12345);
        t1ha1        
---------------------
 1056465972331175422
(1 row)

-- Test integer input with custom seed
SELECT t1ha1(12345, 42);
        t1ha1         
----------------------
 -2965073534604992670
(1 row)

SELECT t1ha1(-12345, 84);
        t1ha1        
---------------------
 3994039927196885025
(1 row)

-- Test consistency (same input should give same hash)
SELECT t1ha1('consistent test') = t1ha1('consistent test');
 ?column? 
----------
 t
(1 row)

-- Test seed effect (same input, different seeds should give different hashes)
SELECT t1ha1('seed test', 1) != t1ha1('seed test', 2);
 ?column? 
----------
 t
(1 row)

-- Test empty string
SELECT t1ha1('');
 t1ha1 
-------
     0
(1 row)

-- Test single character
SELECT t1ha1('a');
        t1ha1        
---------------------
 1862715155098400308
(1 row)

-- Test long string
SELECT t1ha1('Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.');
        t1ha1        
---------------------
 3296404290822654806
(1 row)

-- Test various input lengths to test different code paths
SELECT t1ha1('1234567');           -- 7 bytes
        t1ha1         
----------------------
 -5787676341473605757
(1 row)

SELECT t1ha1('12345678');          -- 8 bytes  
        t1ha1        
---------------------
 1667129700347281224
(1 row)

SELECT t1ha1('123456789012345');   -- 15 bytes
       t1ha1        
--------------------
 774991217029260232
(1 row)

SELECT t1ha1('1234567890123456');  -- 16 bytes
        t1ha1         
----------------------
 -4841296275810522002
(1 row)

SELECT t1ha1('12345678901234567890123456789012'); -- 32 bytes
        t1ha1        
---------------------
 1982017324416562156
(1 row)

SELECT t1ha1('123456789012345678901234567890123'); -- 33 bytes
        t1ha1        
---------------------
 1046741975922950024
(1 row)

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 't1ha1'
ORDER BY proname, proargtypes;
 proname | provolatile | proisstrict 
---------+-------------+-------------
 t1ha1   | i           | t
 t1ha1   | i           | t
 t1ha1   | i           | t
 t1ha1   | i           | t
 t1ha1   | i           | t
 t1ha1   | i           | t
(6 rows)

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';
 extname | extversion 
---------+------------
 hashlib | 0.0.1
(1 row)

